api/src/modules/content-pages/services/content-pages.service.ts

```diff
@@ -419,13 +418,16 @@ export class ContentPagesService {
 				});
 			}
 
-			if (!contentPage.isPublic) {
+			// Because `isPublic` is only set once when publishing a content page we only care about it
+			// if it is the only property governing publication
+			if (!contentPage.publishAt && !contentPage.depublishAt && !contentPage.isPublic) {
```

Comment:
the publishAt and depublishAt should correctly set the isPublic property
so this check should not be needed

The issue is probably located in the webhook that finds all content pages that need to be published/unpublished
otherwise isPublic would be correctly set according to the "publishAt" and "depublishAt"

I'm guessing here:
https://github.com/viaacode/react-admin-core-module/blob/04e4354841bbe7da54c6cb5a064fef19de43433b/api/src/modules/content-pages/services/content-pages.service.ts#L513-L541